test/index.test.js

```diff
@@ -1,94 +1,111 @@
 const { test } = require('tap')
 const Fastify = require('fastify')
 const fastifySlonik = require('../index')
-const { BAD_DB_NAME, connectionString, connectionStringBadDbName } = require('./helpers')
 
-test('Namespace should exist:', (t) => {
-  t.test('fastify.slonik', t => {
-    t.plan(4)
+const connectionString = process.env.DATABASE_URL
+const BAD_DB_NAME = 'db_that_does_not_exist'
+const connectionStringBadDbName = connectionString.replace(/\/[^/]+$/, '/' + BAD_DB_NAME)
 
-    const fastify = Fastify()
-    t.teardown(() => fastify.close())
-
-    fastify.register(fastifySlonik, {
-      connectionString
-    })
-    fastify.ready((err) => {
-      t.error(err)
-
-      t.ok(fastify.slonik)
-      t.ok(fastify.slonik.pool)
-      t.ok(fastify.slonik.connect)
-    })
-  })
-  test('fastify.sql', (t) => {
-    t.plan(2)
+test('Namespace should exist:', async t => {
+  const fastify = Fastify()
 
-    const fastify = Fastify()
-    t.teardown(() => fastify.close())
+  t.teardown(() => fastify.close())
 
-    fastify.register(fastifySlonik, {
-      connectionString
-    })
+  fastify.register(fastifySlonik, { connectionString })
+  await fastify.ready()
 
-    fastify.ready((err) => {
-      t.error(err)
+  t.test('fastify.slonik', async t => {
+    t.ok(fastify.slonik)
+    t.ok(fastify.slonik.pool)
+    t.ok(fastify.slonik.connect)
+    t.ok(fastify.slonik.query)
+    t.ok(fastify.slonik.transaction)
+    t.ok(fastify.slonik.exists)
+  })
 
-      t.ok(fastify.sql)
-    })
+  t.test('fastify.sql', async t => {
+    t.ok(fastify.sql)
   })
   t.end()
 })
 
-test('When fastify.slonik root namespace is used:', (t) => {
-  t.test('should be able to use make a query', t => {
-    t.plan(2)
+test('When fastify.slonik root namespace is used:', async t => {
+  const testName = 'foobar'
 
-    const fastify = Fastify()
-    t.teardown(() => fastify.close())
+  const fastify = Fastify()
 
-    fastify.register(fastifySlonik, {
-      connectionString
-    })
+  t.teardown(async () => {
+    const removeUser = fastify.sql`
+      DELETE FROM
+        users
+      WHERE
+        username=${testName};
+    `
+    await fastify.slonik.transaction(removeUser)
+    fastify.close()
+  })
+
+  fastify.register(fastifySlonik, { connectionString })
+  await fastify.ready()
 
-    fastify.ready(async (err) => {
-      t.error(err)
-      const queryString = fastify.sql`
-        SELECT 1 as one
-      `
+  t.test('should be able to make a query', async t => {
+    const queryString = fastify.sql`
+      SELECT 1 as one
+    `
+    const queryResult = await fastify.slonik.query(queryString)
+    const { rows: [{ one }] } = queryResult
+    t.same(one, 1)
```

Comment:
[t.same](https://node-tap.org/docs/api/asserts/#tsamefound-wanted-message-extra) is intended for deep object comparison, could we switch this to `t.is`? (same with the next test)